hax/hax/handler.py

```diff
@@ -222,9 +225,9 @@ def update_process_failure(self, planner: WorkPlanner,
                             # local motr processes must still be sent.
                             new_ha_states.append(
```

Comment:
Just a question: we broadcast the process ha state even **if its not a local node's or RC node's process**. (outside `if (am_i_rc or is_proc_local):`)
Then again below `if not is_proc_local:` we append the process ha state on line 248.

For **remote processes.**. won't we duplicate it?

Reply:
We update 2 Consul KVs for processes, 
1. processes/{pfid} - This is updated either by the hax on a given node hosting a given process or by RC for remote processes.
2. {node-name}/processes/{pfid} - Here we account remote process's  status with every node, this is to make sure that remote process state changes are received by a given process.

Reply:
Ok Ok.. actually i was talking about this.
```
 Broadcasting HA states [HAState(fid=0x7200000000000001:0x8d, status=OFFLINE), HAState(fid=0x7200000000000001:0x8d, status=OFFLINE)] over ha_link
```
here in logs, I see duplicate ha states... in case of specific processes only.. (remote ones) guess we are appending twice.

Reply:
Oh yeah, duplicate entries are added if the hax process processing a given process event is recovery coordinator and also is a local hax process for that process. We don't do any deduplication as it may affect performance and can easily go to the order of n(square). May be later we can improvise the data structure to be a dictionary instead of a list, this can avoid duplication automatically.